### PR TITLE
Implement support for cal::date_duration

### PR DESCRIPTION
**Pull request by elprans**

### CODE REVIEW

File: edgedb/datatypes/date_duration.pyx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    def __str__(self):
+        # This is closely modeled after interval_out().
+        cdef:
+            int64_t year = self.months / MONTHS_PER_YEAR
+            int64_t mon = self.months % MONTHS_PER_YEAR
+            int64_t tfrac
+            int64_t hour
+            int64_t min
+            int64_t sec
+            int64_t fsec
```
Comment (vpetrovykh) [inline edgedb/datatypes/date_duration.pyx:58]: Do we need these for DateDuration?

File: edgedb/protocol/codecs/codecs.pyx [hunk 2]
```diff
@@ -451,6 +451,35 @@ cdef relative_duration_decode(pgproto.CodecContext settings, FRBuffer *buf):
...
+    buf.write_int64(0)
+    buf.write_int32(days)
+    buf.write_int32(months)
+
+
+cdef date_duration_decode(pgproto.CodecContext settings, FRBuffer *buf):
+    cdef:
+        int32_t days
+        int32_t months
+        int64_t microseconds
```
Comment (vpetrovykh) [inline edgedb/protocol/codecs/codecs.pyx:471]: Why are there microseconds here? Shouldn't the server reliably send the correct data type, the one without fractional day parts?

Comment (elprans) [reply edgedb/protocol/codecs/codecs.pyx:471]: See the assertion below.

File: edgedb/datatypes/date_duration.pyx (hunk 1 above)
Comment (elprans) [reply edgedb/datatypes/date_duration.pyx:58]: Right, these are unused.

File: edgedb/protocol/codecs/codecs.pyx (hunk 2 above)
Comment (elprans) [reply edgedb/protocol/codecs/codecs.pyx:471]: To clarify, there is no "data type without fractions" on the wire, we use `interval` for all durations.

**Review by vpetrovykh — APPROVED** [review]

**Review by 1st1 — APPROVED** [review]

**Review by msullivan — APPROVED** [review]

**Pull request merged**